src/components/modalWindows/addMovie/addMovie.js

```diff
@@ -1,151 +1,193 @@
 import React, { Fragment, useCallback } from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { useFormik } from 'formik';
+import * as Yup from 'yup';
+
+import { addMovie, updateMovie } from './../../../services/requests';
+
+import ErrorMessage from '../../errorMessage';
 
 import './addMovie.css';
 
   const AddMovie = (props) => {
 
     const {
-      id,
-      title,
-      data,
-      genre,
-      url,
-      overview,
-      runtime,
+      activeId,
+      movie,
       handleClose,
-      editMovie
+      editMovie,
+      addNewMovie, 
+      editExistMovie,
     } = props;
-  
-    const handleSignIn = useCallback((event) => {
-      event.preventDefault()
-      const form = event.target;
-      if(!editMovie){
-        const newMovie = {
-          id: `MO${new Date().valueOf()}VIE`,
-          title: form[0].value,
-          data: form[1].value,
-          url: form[2].value,
-          genre: form[3].value,
-          overview: form[4].value,
-          runtime: form[5].value,
-        };
-        // -----TEST FORM ADDMOVIE SUBMIT-----
-        console.log('TEST FORM ADDMOVIE SUBMIT', newMovie)
-        // --------------------------
-      } else {
-        const editMovie = {
-          id: props.id,
-          title: form[0].value,
-          data: form[1].value,
-          url: form[2].value,
-          genre: form[3].value,
-          overview: form[4].value,
-          runtime: form[5].value,
-        };
-        // -----TEST FORM EDITMOVIE SUBMIT-----
-        console.log('TEST FORM EDITMOVIE SUBMIT', editMovie)
-        // --------------------------
-      }
 
-      handleClose();
-    }, [])
+    const formik = useFormik({
+        initialValues: {
+          ...movie
+        },
+        validationSchema: Yup.object({
```

Comment:
Move the validation schema to separate file `move.schema.js`

Reply:
done